cp1616/src/cp1616_io_controller.cpp

```diff
@@ -48,17 +53,29 @@ Cp1616IOController::Cp1616IOController() :
   input_module_count_(0),
   output_module_count_(0),
   input_module_total_data_size_(0),
-  output_module_total_data_size_(0)
+  output_module_total_data_size_(0),
+  num_of_input_modules_(0),
+  num_of_output_modules_(0)
 {
-  //Allocate memory for Input module variables
-  input_module_data_length_.resize(NUM_OF_INPUT_MODULES);
-  input_module_state_.resize(NUM_OF_INPUT_MODULES);
-  input_module_address_.resize(NUM_OF_INPUT_MODULES);
-
-  //Allocate memory for Output module variables
-  output_module_data_length_.resize(NUM_OF_OUTPUT_MODULES);  
-  output_module_state_.resize(NUM_OF_OUTPUT_MODULES);
-  output_module_address_.resize(NUM_OF_OUTPUT_MODULES);
+  //Parse data from yaml config file
+  std::string filepath;
+  nh->getParam("filepath", filepath);
```

Comment:
OK adopted, now only filepath is passed to the constructor. 

My original intention was to use NodeHandle not only to read rosparams but also to create publisher/subscribers in constructor. But I m curious how this can be avoided by using plugin architecture